etl/steps/data/garden/demography/2023-09-27/survivor_percentiles.py

```diff
@@ -0,0 +1,164 @@
+    tb = tb[(tb["type"] == "period") & (tb["format"] == "1x1")]
+    # Drop unused columns
+    tb = tb.drop(columns=["format", "type"])
+
+    # 110+ -> 110
+    log.info("survivor_percentiles: replace 110+ -> 100, set Dtypes.")
+    tb["age"] = tb["age"].replace({"110+": "110"})
+
+    # Dtypes
+    tb = tb.astype(
+        {
+            "year": "Int64",
+            "age": "Int64",
+            "deaths": float,
+            "exposure": float,
+        }
+    )
+
+    # Sort
+    tb = tb.sort_values(["year", "age"])
+
+    # Debugging (pick just one country to speed up calculations)
+    # tb = tb[tb.country == "Denmark"]
+
+    # Actual calculation
+    log.info("survivor_percentiles: calculate surviorship ages (can take some minutes)...")
+    columns_grouping = ["country", "sex", "year"]
+    tb = tb.groupby(columns_grouping).apply(lambda group: obtain_survivorship_ages(group)).reset_index()
+
+    # Unpivot
+    log.info("survivor_percentiles: unpivot")
+    tb = tb.melt(
+        id_vars=["country", "sex", "year"],
+        value_vars=["s1", "s10", "s20", "s30", "s40", "s50", "s60", "s70", "s80", "s90", "s99"],
+        var_name="percentile",
+        value_name="age",
+    )
+    tb = tb.dropna(subset=["percentile"])
+    log.info("survivor_percentiles: rename percentiles (remove 's' and reverse order))")
+    tb["percentile"] = tb["percentile"].str.replace("s", "").astype(int)
+    tb["percentile"] = 100 - tb["percentile"]
+
+    # Dtypes
+    log.info("survivor_percentiles: final dtypes settings.")
+    tb = tb.astype(
+        {
+            "year": int,
+            "percentile": int,
+        }
+    )
+
+    # Set index
+    log.info("survivor_percentiles: set index.")
+    tb = tb.set_index(["country", "year", "sex", "percentile"], verify_integrity=True)
+
+    # Update metadata
+    log.info("survivor_percentiles: metadata.")
+    tb.metadata.short_name = paths.short_name
+    # Propagate origins metadata (unsure why this has not been propagated?)
+    for col in tb.columns:
+        tb[col].metadata.origins = origins
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir, tables=[tb], check_variables_metadata=True, default_metadata=ds_meadow.metadata
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def obtain_survivorship_ages(tb_group: Table, start_age: int = 0, end_age: int = 110) -> pd.DataFrame:
+    """Get survivorship ages given a life and deaths table.
+
+    Output dataframe has a column for each percentile of survivorship age.
+
+    tb_group is expected to be a subset of the compelte table. It should only concern a particular (country, year, sex) triple.
+    """
+    # Step 1: Apply splines, get Mx for each (country, year, sex, age)
+    ## Define splines
+    spline_deaths = UnivariateSpline(tb_group["age"], tb_group["deaths"], s=0, k=3)
```

Comment:
Cool! I wonder if it's just worth noting what the s and k mean here. From [documentation](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.UnivariateSpline.html):

If 0, spline will interpolate through all data points. This is equivalent to [InterpolatedUnivariateSpline](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.InterpolatedUnivariateSpline.html#scipy.interpolate.InterpolatedUnivariateSpline).

And: 
k = 3 is a cubic spline.

I was wondering if the function would be equivalent across R and Python and I think this is a key point of clarification on that.